posts/2024-05-03-simplifying-nosql-database-integration-with-jakarta-nosql.adoc

```diff
@@ -0,0 +1,417 @@
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/mongodb.png[MongoDB ,align="left" width=25%,height=25%]
+[source, java]
+----
+var document = new Document();
+document.append(name, value);
+----
+****
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/ArangoDB.png[Arango DB,align="left" width=25%,height=25%]
+[source, java]
+----
+var baseDocument = new BaseDocument();
+baseDocument.addAttribute(name, value);
+----
+****
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/couchbase.svg[Couchbase,align="left" width=25%,height=25%]
+[source, java]
+----
+var jsonObject = JsonObject.create();
+jsonObject.put(name, value);
+----
+****
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/orientdb.png[Orient Project,align="left" width=25%,height=25%]
+[source, java]
+----
+var document = new ODocument("collection");
+document.field(name, value);
+----
+****
+
+With Eclipse JNoSQL, developers can use a common API to integrate with different database types, free of vendor lock-in, and with a low cognitive load during learning phases. For example, using the Document API, it's possible to switch between MongoDB and ArangoDB as needed, based on Convention Over Configuration (CoC).
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/images/home_logo.png[Eclipse JNoSQL Project,align="left" width=25%,height=25%]
+[source, java]
+----
+var entity = CommunicationEntity.of("collection");
+entity.add(name, value);
+----
+****
+
+Check out some of the Jakarta NoSQL annotations:
+
+[source, java]
+----
+import jakarta.nosql.Entity;
+import jakarta.nosql.Id;
+import jakarta.nosql.Column;
+
+@Entity
+public class Book {
+
+    @Id
+    private String isbn;
+
+    @Column
+    private String title;
+
+    @Column
+    private String author;
+
+    @Convert(YearConverter.class)
+    @Column
+    private Year year;
+
+}
+----
+
+When using Java 17+, Eclipse JNoSQL allows the usage of Java Records as entities:
+
+[source, java]
+----
+import jakarta.nosql.Entity;
+import jakarta.nosql.Id;
+import jakarta.nosql.Column;
+
+@Entity
+public record Book(@Id String isbn,
+                   @Column("title") String title,
+                   @Column("author") String author,
+                   @Convert(YearConverter.class) @Column("year") Year year,
+                   @Column("edition") int edition) {
+
+}
+
+----
+
+Last but not least, Eclipse JNoSQL, as a Jakarta Data implementation, allows us to create repositories, enabling Domain-Driven Development (DDD) capabilities through the usage of the Repository pattern. This approach simplifies developers who seek to bring the code closer to the business (domain-centric) instead of working with database semantics.
```

Comment:
```suggestion
Last but not least, Eclipse JNoSQL, as a Jakarta Data implementation, allows us to create repositories, enabling domain-driven development (DDD) capabilities through the usage of the repository pattern. This approach helps developers bring the code closer to the business (domain-centric) instead of working with database semantics.
```